lib/netext/httpext/request.go

```diff
@@ -187,13 +181,18 @@ func MakeRequest(ctx context.Context, state *lib.State, preq *ParsedHTTPRequest)
 		}
 	}
 
-	tracerTransport := newTransport(ctx, state, tags, preq.ResponseCallback)
+	tracerTransport := newTransport(ctx, state, &preq.TagsAndMeta, preq.ResponseCallback)
 	var transport http.RoundTripper = tracerTransport
 
 	if state.Options.HTTPDebug.String != "" {
 		// Combine tags with common log fields
 		combinedLogFields := map[string]interface{}{"source": "http-debug", "vu": state.VUID, "iter": state.Iteration}
-		for k, v := range tags.Map() {
+		for k, v := range preq.TagsAndMeta.Tags.Map() {
```

Comment:
Maybe create a small method `TagsAndMeta.All() map[string]` to consolidate the access/management inside the `TagsAndMeta` struct and expose less outside? WDYT?

Reply:
Tried this, but I'd prefer to leave things as they are. In most places, this helper method will just lead to more allocations :disappointed: Here is, but also in these other places: https://github.com/grafana/k6/blob/d3d0eb8af4b14a169143126a0f839f1f6833b153/output/csv/output.go#L238-L247
https://github.com/grafana/k6/blob/d3d0eb8af4b14a169143126a0f839f1f6833b153/output/json/wrapper.go#L46
https://github.com/grafana/k6/blob/d3d0eb8af4b14a169143126a0f839f1f6833b153/js/modules/k6/execution/execution.go#L379-L386

As far as I can see, there is no single place where such a method won't cause unnecessary extra allocations. And since it can be so easily misused, I'd prefer to not add it